decoder.go

```diff
@@ -95,24 +95,22 @@ func (d *decoder) next() {
 		d.buffered = ""
 	}
 
-	if len(d.current) > 65 {
-		isContinuation := true
-		for isContinuation == true {
-			res := d.scanner.Scan()
-			d.line++
-			if true != res {
-				d.err = d.scanner.Err()
-				return
-			}
-			line := d.scanner.Text()
-			if strings.HasPrefix(line, " ") || strings.HasPrefix(line, "\t") {
-				d.current = d.current + line[1:]
-			} else {
-				// If is not a continuation line, buffer it, for the
-				// next call.
-				d.buffered = line
-				isContinuation = false
-			}
+	for {
+		res := d.scanner.Scan()
+		if true != res {
+			d.err = d.scanner.Err()
+			// there is nothing more to scan
+			break
+		}
+		d.line++
```

Comment:
I'm not super familiar with this code, but the original increments `d.line` before breaking out of the loop. I have a hunch it's going to throw off `(*decoder) Lines()` for every file. I think your version is more correct, and there should probably be a test for it.

Reply:
Hum after all the `func Lines()` comment says:

>  // Lines processed. If Decoder reports an error.

So if I understand, it should return the line number of the error. So with my changes, the line number returned is only the last good one.
I suppose I need to increment `d.line` if the read has an error.

My test causes a panic with the `master` branch